### PR TITLE
Boto

### LINKED ISSUE #20: exclude_zero_byte_files - error in `copy_s3_folder_contents_to_new_folder`
https://github.com/moj-analytical-services/dataengineeringutils3/blob/81d2b193ff9ecf428d4efb7064db7114e261b2c9/dataengineeringutils3/s3.py#L129

Hard coded rather than set by input param for function

### PR DESCRIPTION
**Pull request by isichei**
## Loosening boto3 restriction

- closes #20
- Updated test to make it more readable
- Updated tests to work with later mocked S3 create bucket call
- Split out current workflow into two workflows (testing and linting)

Everything else is cosmetic

### CODE REVIEW

File: dataengineeringutils3/s3.py [hunk 1]
```diff
@@ -126,7 +130,7 @@ def copy_s3_folder_contents_to_new_folder(
     to_s3_folder_path = _add_slash(to_s3_folder_path)
 
     all_from_filepaths = get_filepaths_from_s3_folder(
-        from_s3_folder_path, exclude_zero_byte_files=False
+        from_s3_folder_path, exclude_zero_byte_files=exclude_zero_byte_files
```
Comment (isichei) [inline dataengineeringutils3/s3.py:133]: Actual change.

File: pyproject.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "dataengineeringutils3"
-version = "1.1.0"
+version = "1.1.1"
```
Comment (isichei) [inline pyproject.toml:3]: Actual change.

File: pyproject.toml [hunk 3]
```diff
@@ -18,7 +18,7 @@ classifiers = [
 
 [tool.poetry.dependencies]
 python = "^3.6"
-boto3 = "^1.10"
+boto3 = "^1"
```
Comment (isichei) [inline pyproject.toml:21]: Actual change.

File: tests/test_integration.py [hunk 4]
```diff
@@ -21,7 +19,10 @@ def test_large_select_queryset_with_writer(s3, large_select_queryset):
     """
     Test file writer and queryset with large result set.
     """
-    s3.meta.client.create_bucket(Bucket=BUCKET_NAME)
+    s3.meta.client.create_bucket(
+        Bucket=BUCKET_NAME,
+        CreateBucketConfiguration={"LocationConstraint": "eu-west-1"},
```
Comment (isichei) [inline tests/test_integration.py:24]: Actual change.

This was done throughout. Without `LocationConstraint` all tests were failing.

File: tests/test_s3.py [hunk 5]
```diff
@@ -127,42 +139,59 @@ def test_write_json_s3(s3):
...
 
-def test_copy_s3_folder_contents_to_new_folder(s3):
-
-    files = [
-        {"folder": "f1", "key": "my_file.json", "body": "test"},
-        {"folder": "f1", "key": "df.first.py", "body": "test"},
-        {"folder": "f1", "key": "otherfile.json", "body": ""},
-        {"folder": "f.2", "key": "ffile.json", "body": "test"},
-        {"folder": "f.2", "key": "otherfile.json", "body": "test"},
+@pytest.mark.parametrize(
```
Comment (isichei) [inline tests/test_s3.py:142]: Actual change. (this whole function)

**Review by gkelly900 — APPROVED** [review]

**Pull request merged**